src/service/repo/signedMessageRepo.ts

```diff
@@ -44,8 +59,17 @@ async function checkIsTargetAllowingAnonDM(targetUserId: string) {
   }
 }
 
-async function sendSignedMessage(channelId: string, message: string, channelType: number) {
-  const payload = {channelId, message, channelType};
+async function sendSignedMessage(
+  channelId: string,
+  message: string,
+  channelType: number,
+  replyMessageId?: string
+) {
+  let payload: SendPayloadType = {channelId, message, channelType};
+  if (replyMessageId) {
+    payload = {...payload, messageType: 'reply', replyMessageId};
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `sendSignedMessage` function has been updated to handle optional reply messages. The logic seems correct: if a `replyMessageId` is provided, it's added to the payload along with a `messageType` of 'reply'. However, I would suggest adding error handling in case the `channelId`, `message`, or `channelType` are not provided, as these are required fields.

```diff
async function sendSignedMessage(
  channelId: string,
  message: string,
  channelType: number,
  replyMessageId?: string
) {
  if (!channelId || !message || !channelType) {
    throw new Error('Required parameters are missing');
  }
  let payload: SendPayloadType = {channelId, message, channelType};
  if (replyMessageId) {
    payload = {...payload, messageType: 'reply', replyMessageId};
  }

```



<!-- This is an auto-generated comment by OSS CodeRabbit -->